### PR TITLE
Openjdk tomcat ubuntu vm 

### PR DESCRIPTION
**Pull request by leestott**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by leestott** [comment]
@bmoore-msft  can you please approve this PR

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json [hunk 1]
```diff
@@ -1,251 +1,296 @@
...
-        "dnsSettings": {
-          "domainNameLabel": "[parameters('vmDnsName')]"
+      "ubuntuOSVersion": {
+        "type": "string",
+        "defaultValue": "18.04-LTS",
+        "allowedValues": [
+          "18.04-LTS"
+        ],
+        "metadata": {
+          "description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version. Allowed values: 12.04.5-LTS, 14.04.5-LTS, 15.10."
```
Comment (bmoore-msft) [inline openjdk-tomcat-ubuntu-vm/azuredeploy.json]: this doesn't match the allowedValues - but in general doesn't make sense that this is a param when the offer/sku are not...

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json [hunk 2]
```diff
@@ -1,251 +1,296 @@
...
+                      "direction": "Inbound"
+                    }
+                  },
+                  {
+                    "name": "RDP",
+                    "properties": {
+                      "description": "Allow RDP",
+                      "protocol": "Tcp",
+                      "sourcePortRange": "*",
+                      "destinationPortRange": "3389",
```
Comment (bmoore-msft) [inline openjdk-tomcat-ubuntu-vm/azuredeploy.json]: why 3389?

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json [hunk 3]
```diff
@@ -1,251 +1,296 @@
...
+          },
+          "storageProfile": {
+            "imageReference": {
+              "publisher": "[variables('imagePublisher')]",
+              "offer": "[variables('imageOffer')]",
+              "sku": "[parameters('ubuntuOSVersion')]",
+              "version": "latest"
+            },
+            "osDisk": {
+              "osType": "Linux",
```
Comment (bmoore-msft) [inline openjdk-tomcat-ubuntu-vm/azuredeploy.json]: remove this property

File: openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json [hunk 4]
```diff
@@ -3,16 +3,10 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
     "javaPackageName": {
-      "value": "openjdk-7-jdk"
+      "value": "openjdk-13-jdk"
```
Comment (bmoore-msft) [inline openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json]: remove this and use a defaultValue in the template

File: openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json [hunk 5]
```diff
@@ -3,16 +3,10 @@
...
     },
     "tomcatPackageName": {
-      "value": "tomcat7"
-    },
-    "newStorageAccountName": {
-      "value": "GEN-UNIQUE-8"
-    },
-    "vmDnsName": {
-      "value": "GEN-UNIQUE-8"
+      "value": "tomcat9"
```
Comment (bmoore-msft) [inline openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json]: remove this and use a defaultValue in the template

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json (hunk 1 above)
Comment (leestott) [reply openjdk-tomcat-ubuntu-vm/azuredeploy.json]: Ok Added 16.04-LTS and 14.04.5-LTS as possible versions

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json (hunk 2 above)
Comment (leestott) [reply openjdk-tomcat-ubuntu-vm/azuredeploy.json]: Default RDP port for Windows and Linux

File: openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json (hunk 4 above)
Comment (leestott) [reply openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json]: yes added default "

defaultValue": "openjdk-13-jdk",
            "allowedValues": [
              "openjdk-11-jdk",
              "openjdk-12-jdk",
              "openjdk-13-jdk"

File: openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json (hunk 5 above)
Comment (leestott) [reply openjdk-tomcat-ubuntu-vm/azuredeploy.parameters.json]: yes completed    

"tomcatPackageName": {
            "type": "string",
            "defaultValue": "tomcat9",
           "allowedValues": [
              "tomcat7",
              "tomcat8",
              "tomcat9"
            ],
            "metadata": {
              "description": "tomcat Package Name"
            }
          },

**Comment by leestott** [comment]
@bmoore-msft  updated based on feedback

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json (hunk 3 above)
Comment (leestott) [reply openjdk-tomcat-ubuntu-vm/azuredeploy.json]: Added  
            "osDisk": {
              "osType": "Linux",
              "name": "[concat(variables('vmName'),'_OSDisk')]",
              "caching": "ReadWrite",
              "createOption": "FromImage",
              "managedDisk": {
                "storageAccountType": "Premium_LRS"
              }
            }

File: openjdk-tomcat-ubuntu-vm/azuredeploy.json [hunk 6]
```diff
@@ -1,251 +1,284 @@
...
+              "name": "[concat(variables('vmName'),'_OSDisk')]",
+              "caching": "ReadWrite",
+              "createOption": "FromImage",
+              "managedDisk": {
+                "storageAccountType": "Premium_LRS"
```
Comment (bmoore-msft) [inline openjdk-tomcat-ubuntu-vm/azuredeploy.json]: change this to Standard_SSD - that's supported for all vm sizes

Comment (leestott) [reply openjdk-tomcat-ubuntu-vm/azuredeploy.json]: Updated to     "storageAccountType": "StandardSSD_LRS"

**Pull request merged**